lib/rubocop/cop/autocorrect_logic.rb

```diff
@@ -0,0 +1,22 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    # This module encapsulates the logic for autocorrect behaviour for a cop
+    module AutocorrectLogic
+      def autocorrect?
```

Comment:
On the other hand it's some separation of concerns, similar to what we did for the `Utils`. Design is hard... :-)